### PR TITLE
Mobile local dev: Unity project changes + IL2CPP correction

### PR DESCRIPTION
**Pull request by gkassabli**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR adds Unity project settings changes to support iOS and Android. It also corrects IL2CPP assembly linking that is used in iOS.
#### Tests
Tested along with other PRs in `mobile_merge` on iOS and Android devices and simulators
#### Primary reviewers
cc @DanailPenev

### CODE REVIEW

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -546,8 +639,8 @@ PlayerSettings:
   webGLLinkerTarget: 0
   scriptingDefineSymbols:
     1: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP
-    4: UNITY_POST_PROCESSING_STACK_V2
-    7: UNITY_POST_PROCESSING_STACK_V2
+    4: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP
+    7: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP
```
Comment (samiwh) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: Why is this added back in

Comment (gkassabli) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: I think this was accidental, I'll revert this

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 2]
```diff
@@ -546,8 +639,8 @@ PlayerSettings:
   webGLLinkerTarget: 0
   scriptingDefineSymbols:
     1: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP
-    4: UNITY_POST_PROCESSING_STACK_V2
-    7: UNITY_POST_PROCESSING_STACK_V2
+    4: UNITY_POST_PROCESSING_STACK_V2;
```
Comment (jessicafalk) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: this looks like it needs the `UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP` symbol

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 1 above)
Comment (DanailPenev) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: @samiwh what do you mean back in? It's still present in master for PC. This just adds the symbol for Android and iOS.

Comment (jessicafalk) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: why did this got removed?

Comment (jessicafalk) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: agreeing with @DanailPenev  here, we need these symbols for mobile support.

Comment (samiwh) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: Out of interest what breaks then if you don't enable the automatic bootstrap on mobile?

Comment (jessicafalk) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: none of the systems in the playground are annotated with `DisableAutoCreation`. You might remember the discussion we had back then when deciding whether to have a default world or not. Those will be added to the default world. There are a few systems in the playground that also use the WorkerSystem. Those will throw errors without that symbol

Comment (jessicafalk) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: in general: Either we have the automatic bootstrap enabled for all platforms or for none. Having it only for a few just seems wrong and could introduce issues without us realizing it

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 3]
```diff
@@ -47,7 +47,7 @@ PlayerSettings:
   defaultScreenWidthWeb: 960
   defaultScreenHeightWeb: 600
   m_StereoRenderingPath: 0
-  m_ActiveColorSpace: 1
```
Comment (firtina-improbable) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: Is this intentional?

Comment (gkassabli) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: Yes, without this change either iOS Sim or iOS device (don't remember which exactly) doesn't work correctly.

Comment (firtina-improbable) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: 👍

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 4]
```diff
@@ -150,7 +150,8 @@ PlayerSettings:
   resolutionScalingMode: 0
   androidSupportedAspectRatio: 1
   androidMaxAspectRatio: 2.1
-  applicationIdentifier: {}
+  applicationIdentifier:
+    Android: com.improbable.gdk
```
Comment (jessicafalk) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: while you are on that. could you also fix that for the other platforms? I believe the standalone and ios player settings use the default one `com.Company.ProductName`....

should really change that 😂

Comment (gkassabli) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: (y)
